### PR TITLE
Logout security

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
This pr checks if the logout redirect is a `*.dosomething.org` domain so a third party cant' hijack a users experience.

#### How should this be reviewed?
Try going to `/logout?redirect="http://heyheyhey.com"` and `/logout?redirect="http://dosomething.org"`

*note: the HTTP part matters. if we have apps not including http in the redirect url they need to be updated as well.

**Fixes**: https://www.pivotaltracker.com/story/show/150597261

#### Checklist
- [x] Tests added for new features/bug fixes.

### CODE REVIEW

File: app/helpers.php [hunk 1]
```diff
@@ -185,3 +191,20 @@ function format_legacy_mobile($mobile)
...
+ */
+function is_dosomething_domain($url)
+{
+    $parsed = parse_url($url);
+
+    if (! array_key_exists('host', $parsed)) {
+        return false;
+    }
+
+    return strpos($parsed['host'], 'dosomething.org') !== false;
```
Comment (DFurnes) [inline app/helpers.php]: Let's use [`ends_with`](https://laravel.com/docs/5.3/helpers#method-ends-with) here to make sure this wouldn't allow someone to pass a domain like `dosomething.org.evil.herokuapp.com`.

File: app/helpers.php [hunk 2]
```diff
@@ -173,6 +173,12 @@ function format_birthdate($birthdate)
     return format_date($birthdate, 'Y-m-d');
 }
 
+/**
+ * Format a legacy phone number to a proper number format.
```
Comment (DFurnes) [inline app/helpers.php]: 👏 🙇

File: app/helpers.php (hunk 1 above)
Comment (itsjoekent) [reply app/helpers.php]: @DFurnes Updated & also updated the test to mimic this

File: tests/Http/Web/WebAuthenticationTest.php [hunk 3]
```diff
@@ -154,6 +154,25 @@ public function testLogoutRedirect()
...
+     * Test that we can't be redirected to a third party domain
+     * in the custom post-logout redirect.
+     */
+    public function testLogoutRedirectThirdPartyDomain()
+    {
+        $user = factory(User::class)->create();
+
+        $this->be($user, 'web');
+
+        $this->get('logout?redirect=http://puppet.sloth.com');
```
Comment (DFurnes) [inline tests/Http/Web/WebAuthenticationTest.php]: Can you update the query param here to match the new assertion you're making below?

Comment (itsjoekent) [reply tests/Http/Web/WebAuthenticationTest.php]: Oh yes, good catch 🌮

**Pull request merged**